annotations/manifest_upload.py

```diff
@@ -71,13 +71,14 @@ def edit_manifest(file_path, col_types_dict):
     # Adjust data types to match table schema
     for columnName in df:
         if col_types_dict[columnName] == list:
+            df[columnName] = df[columnName].astype(str)
             df[columnName] = df[columnName].str.split(', ')
         else:
             for k, v in col_types_dict.items():
```

Comment:
This loop is probably not necessary, since you're only looking at one column at a time. And since a column can only have one type, an iteration through the dict is a little superfluous.  You're also not using `k` or `v` at all, which is another hint that the loop is not needed!

So rather than this, I suggest looking up the type directly:

```suggestion
            df[columnName] = df[columnName].astype(
                col_types_dict[columnName])
```

Reply:
Also, does `col_type_dict` contain all the of column types we need? If not, I suggest using `get()` so that we can set a default type, e.g. str, if a key is not found:

```
            df[columnName] = df[columnName].astype(
                col_types_dict.get(columnName, str))
```

Reply:
Ah! Thank you for catching this!